### PR TITLE
chore: clean up remaining ethers usage

### LINKED ISSUE #6252: `cast interface` does not include payable address modifier
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (85f25c6 2023-11-08T00:33:45.948577000Z)

### What command(s) is the bug in?

cast interface

### Operating System

macOS (Intel)

### Describe the bug

The interface generated by `cast` does not include the `payable` modifier on address params.

This is non-critical since values can always be casted, but it makes integrations more difficult when using the generated interface and expecting a payable return param (or a `address payable[] memory`). 

Example contract:
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract Payable {
  function getPayable() external view returns (address payable) {
    return payable(address(this));
  }

  function setPayable(address payable test) external pure {
    test;
  }
}
```

`cast interface <(forge inspect Payable abi)` outputs:
```solidity
pragma solidity ^0.8.10;

interface Interface {
    function getPayable() external view returns (address);
    function setPayable(address test) external pure;
}
```

Expected:
```solidity
pragma solidity ^0.8.10;

interface Interface {
    function getPayable() external view returns (address payable);
    function setPayable(address payable test) external pure;
}
```

### PR DESCRIPTION
**Pull request by DaniPopes**
Fixes #6252

### CODE REVIEW

File: crates/config/src/cache.rs
Comment (DaniPopes) [inline crates/config/src/cache.rs:1]: `-️` -> `-`

File: crates/cast/bin/cmd/interface.rs [hunk 1]
```diff
@@ -19,7 +20,7 @@ pub struct InterfaceArgs {
     name: Option<String>,
 
     /// Solidity pragma version.
-    #[clap(long, short, default_value = "^0.8.10", value_name = "VERSION")]
+    #[clap(long, short, default_value = "^0.8.4", value_name = "VERSION")]
```
Comment (mattsse) [inline crates/cast/bin/cmd/interface.rs:23]: any particular reason for this?

File: crates/config/src/cache.rs
Comment (mattsse) [reply crates/config/src/cache.rs:1]: makes sense

**Review by mattsse — APPROVED** [review]
monster PR, 
luckily most if it are rename/import changes, 
actual changes lgtm
one question re default solc pragma

File: crates/cast/bin/cmd/interface.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/cast/bin/cmd/interface.rs:23]: custom errors were introduced in 0.8.4

**Pull request merged**